engine/src/test/java/org/camunda/bpm/engine/test/bpmn/event/compensate/helper/CompensationConceptsTests.java

```diff
@@ -0,0 +1,490 @@
+        Map<String, Object> variables = taskService.getVariables(collectPointsTask.getId());
+
+        assertEquals(1, variables.size());
+        assertEquals("false", variables.get("isVital"));
+
+
+        // trigger error on non-vital task
+        taskService.handleBpmnError(collectPointsTask.getId(), "errorCode");
+
+        // verify that NonVital Task has been completed
+        assertNull(taskService.createTaskQuery().taskName("Collect Royality Points").singleResult());
+        historicActivityInstance =
+                historyService.createHistoricActivityInstanceQuery().orderByHistoricActivityInstanceStartTime().asc().list();
+        assertEquals(4, historicActivityInstance.size()); // start Event, bookFlight, collectPoints
+        assertNotNull(historicActivityInstance.get(2).getEndTime()); // 2nd as 3rd is taken by payFlight
+
+        // Non-vital Task should be completed/finished
+        collectPointsTask = taskService.createTaskQuery().taskName("Collect Royality Points").singleResult();
+        assertNull(collectPointsTask);
+
+        assertEquals("Created",
+                taskService.createTaskQuery().taskDefinitionKey("payFlight").singleResult().getTaskState());
+        completeTask("Pay Flight");
+        testRule.assertProcessEnded(processInstanceId);
+    }
+
+    @Deployment(resources = "org/camunda/bpm/engine/test/bpmn/event/compensate/CompensationConceptsTest.retryTaskTest" +
+            ".bpmn20.xml")
+    @Test
+    public void retryTaskTest() {
+        // Tests, that a retry task retries actually failed task and important task parameters are updated. Also
+        // tests if retryCooldown is waited for.
+        String processInstanceId = runtimeService.startProcessInstanceByKey("flightBookingProcess").getId();
+        completeTask("Book Flight");
+
+        // check if task is completed
+        Task bookFlightTask = taskService.createTaskQuery().taskName("Book Flight").singleResult();
+        assertNull(bookFlightTask);
+
+        // check if retry task is activated
+        assertEquals("Created",
+                taskService.createTaskQuery().taskDefinitionKey("payFlight").singleResult().getTaskState());
+
+        // check if inputVariables on retry task are present
+        Task payFlightTask = taskService.createTaskQuery().taskName("Pay Flight").singleResult();
+        Map<String, Object> variables = taskService.getVariables(payFlightTask.getId());
+
+        assertEquals(1, variables.size());
+        TreeMap<String, String> variableMap = (TreeMap<String, String>) variables.get("isRetryTask");
+        assertEquals("true", variableMap.get("isRetryTask"));
+        assertEquals("3", variableMap.get("retryCount"));
+        assertEquals("1", variableMap.get("retryCooldown"));
+
+
+        // get prev start time
+        Date beforeErrorCreateTime =
+                taskService.createTaskQuery().taskName("Pay Flight").singleResult().getCreateTime();
```

Comment:
This should be simple Date.now() as the task has been created already long before this.